src/machine/machinectl.c

```diff
@@ -505,18 +580,31 @@ static int print_addresses(sd_bus *bus, const char *name, int ifi, const char *p
                 if (r < 0)
                         return bus_log_parse_error(r);
 
-                fputs(prefix, stdout);
-                fputs(inet_ntop(family, a, buffer, sizeof(buffer)), stdout);
-                if (family == AF_INET6 && ifi > 0)
-                        printf("%%%i", ifi);
-                fputc('\n', stdout);
+                if (n_addr != 0) {
+                        if (family == AF_INET6 && ifi > 0)
+                                xsprintf(buf_ifi, "%%%i", ifi);
+                        else
+                                strcpy(buf_ifi, "");
+
+                        tmp = strjoin(addresses, prefix,
+                                inet_ntop(family, a, buffer, sizeof(buffer)),
+                                buf_ifi, NULL);
+                        addresses = mfree(addresses);
+                        if (!tmp)
+                                return log_oom();
+                        addresses = tmp;
+                } else
```

Comment:
Sorry I didn't notice this earlier, but we actually have strextend() which makes the above a lot shorter.